crates/gen-host-js/src/lib.rs

```diff
@@ -74,20 +41,55 @@ struct Js {
     all_intrinsics: BTreeSet<Intrinsic>,
 }
 
-#[derive(Default, Debug, Clone)]
+#[derive(Debug, Clone, Default)]
 #[cfg_attr(feature = "clap", derive(clap::Args))]
 pub struct Opts {
     /// Disables generation of `*.d.ts` files and instead only generates `*.js`
     /// source files.
     #[cfg_attr(feature = "clap", arg(long = "no-typescript"))]
     pub no_typescript: bool,
+    /// Provide a custom JS instantiation API for the component instead
+    /// of the direct importable native ESM output.
+    #[cfg_attr(
+        feature = "clap",
+        arg(
+            long = "instantiation",
+            short = 'I',
+            conflicts_with = "compatibility",
+            conflicts_with = "compat"
+        )
+    )]
+    pub instantiation: bool,
+    /// Comma-separated list of "from-specifier=./to-specifier.js" mappings of
+    /// component import specifiers to JS import specifiers.
+    #[cfg_attr(feature = "clap", arg(long = "map"))]
+    pub map: Option<String>,
+    /// Enables all compat flags: --nodejs-compat.
+    #[cfg_attr(feature = "clap", arg(long = "compat"))]
+    pub compat: bool,
+    /// Enables compatibility in Node.js without a fetch global.
+    #[cfg_attr(
+        feature = "clap",
+        arg(
+            long = "nodejs-compat",
+            group = "compatibility",
+            conflicts_with = "compat"
+        )
+    )]
+    pub nodejs_compat: bool,
 }
 
 impl Opts {
-    pub fn build(self) -> Box<dyn ComponentGenerator> {
+    pub fn build(self) -> Result<Box<dyn ComponentGenerator>> {
         let mut gen = Js::default();
         gen.opts = self;
-        Box::new(gen)
+        if gen.opts.compat {
+            gen.opts.nodejs_compat = true;
+        }
+        if let Some(map) = gen.opts.map.as_ref() {
+            gen.import_maps = maps_str_to_map(&map)?;
```

Comment:
I think that you can use `#[clap(value_parser = maps_str_to_map)]` on the field in `Opts` to avoid needing to manually do the parse step here if you'd like.